test/kerodon/unit/core.clj

```diff
@@ -335,6 +335,22 @@
       (test-fill-in-helper '([:label {:for "message-id"} "Message"]
                              [:textarea {:id "message-id" :name "message"}])
                            "Message" :#message-id "blah" "message=blah"))
+    (testing "multiple array inputs"
+      ;; not using label since doing so requires id, and having an id masks the problem
+      ;; from issue #47 when multiple inputs share the same name:
+      (let [state {:app (constantly :x)
+                   :enlive (parse [:form {:action "/" :method "post"}
+                                   [:input.array-input {:name "something"}]
+                                   [:input.array-input {:name "something"}]
+                                   [:input {:type "submit" :value "Submit"}]])}
+            selectors (map (fn [n]
+                               [n [[:.array-input (enlive/nth-of-type n)]]])
```

Comment:
Could you unroll this `(map)` and `(reduce)` into two function calls with `(->)`? it should make it a bit clearer what the test is doing. You could either keep the nth logic out in a local helper called something like `nth-selector`, or just inline it and let it repeat.